### ISSUE TITLE
Неустранимая ошибка 10013

### ISSUE THREAD

**Issue by igvogost** [deemru/Chromium-Gost#60]
Запускаю Chromium-Gost 106.0.5249.103 и в Просмотре событий вижу 10 ошибок на таких ОС:
Win7x32: Произошла неустранимая ошибка при создании учетных данных SSL client. Внутреннее состояние ошибки: 10013
Win10x64 LTSB: Произошла неустранимая ошибка при создании учетных данных TLS client. Внутреннее состояние ошибки: 10013
Win10x64 LTSC: Произошла неустранимая ошибка при создании учетных данных TLS client. Внутреннее состояние ошибки: 10013
Пробовал запускать Chromium 106.0.5249.103 и Google Chrome 106.0.5249.103. Ошибки не появляются.
В итоге откатился на Chromium-Gost 89.0.4389.128. При его запуске и работе ошибки не появляются.

**Comment by deemru** [comment]
Вы не указали вашу версию КриптоПро CSP.

Если это воспроизводится на чистой системе, опишите полный порядок действий для достижения ошибки.

**Comment by deemru** [comment]
> Остальные ОС без КриптоПро CSP и ошибка возникает сразу после загрузки ОС и запуска Chromium-Gost.

Звучит как троллинг. Без КриптоПро CSP ГОСТ работать не должен.

Или проблема только в том что в событиях Windows ошибки, но в целом все работает без ошибок?

**Comment by VolkovAnton83** [comment]
> Это ожидаемые ошибки "простукивания" системного TLS функционала при старте браузера, от чего далее зависит активация тех или иных функций внутри браузера.

Так вы их "простукивайте" начиная с самых новых по направлению к старым. И если самый новый находится, то на нём и работайте...

**Comment by deemru** [comment]
> Так вы их "простукивайте" начиная с самых новых по направлению к старым. И если самый новый находится, то на нём и работайте...
> 

А если на сервере он не поддерживается?

**Comment by VolkovAnton83** [comment]
> А если на сервере он не поддерживается?

Ну, разумеется работать на том, который самый новый из поддерживаемых сервером. Имхо, слабая безопасность на сервере встречается реже, чем на клиенте.

**Comment by deemru** [comment]
> Ну, разумеется работать на том, который самый новый из поддерживаемых сервером. Имхо, слабая безопасность на сервере встречается реже, чем на клиенте. 
> 

Вероятно вы не обратили внимание, хотя процитировали мой изначальный ответ, что простукивание осуществляется разово "при старте браузера".

Узнать при старте браузера по какими серверами будет ходить пользователь невозможно.

**Comment by VolkovAnton83** [comment]
> Вероятно вы не обратили внимание, хотя процитировали мой изначальный ответ, что простукивание осуществляется разово "при старте браузера".

Да, признаюсь, не обратил. Тогда, конечно, надо проверять возможность работы всех протоколов. Впрочем, во-первых, в процессе работы пользователь может "включить" или "отключить" возможность работы протоколов, а во-вторых, почему для определения включенных протоколов не используете ключ реестра Компьютер\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Internet Settings\SecureProtocols?

**Comment by deemru** [comment]
> во-первых, в процессе работы пользователь может "включить" или "отключить" возможность работы протоколов, а во-вторых, почему для определения включенных протоколов не используете ключ реестра

1) Всё так. Может. Но жалоб не было.
2) Абстракция msspi даёт больше гарантий работы для всех поддерживаемых систем. Разбираться в сочетаниях ключей реестра одной конкретной системы выглядит нецелесообразным.

**Comment by VolkovAnton83** [comment]
> Если найдется способ без оставления ошибок в событиях Windows, возьмем на вооружение. Пока так.

> 2\. Разбираться в сочетаниях ключей реестра одной конкретной системы выглядит нецелесообразным.

**Comment by deemru** [comment]
> > Если найдется способ без оставления ошибок в событиях Windows, возьмем на вооружение. Пока так.
> 
> > 2. Разбираться в сочетаниях ключей реестра одной конкретной системы выглядит нецелесообразным.

Да, спасибо за направление, но этот вариант не подходит.

**Comment by VolkovAnton83** [comment]
Если это ещё актуально вообще, но в файле msspi.cpp у вас встречается структура SCHANNEL_CRED, которая, согласно информации на странице описания (https://learn.microsoft.com/en-us/windows/win32/api/schannel/ns-schannel-schannel_cred) не рекомендуется. Вместо нее рекомендуют SCH_CREDENTIALS. Не думаю, что проблема в этом, но всё же ошибка в журнале при проверке работоспособности протокола - это как-то неаккуратненько...

**Comment by karabelnikov** [comment]
Если кому интересно, вывод этих ошибок в системный журнал, можно просто отключить.

**Решение**:
Открыть просмотр событий. Нажать на это событие, там справа - подробнее. Раскрыть System и там видно GuID.
Выглядит примерно так (у всех он разный) - c26c4f3c-3f66-4e99-8f8a-39405cfed220

Открыть реестр и искать тут: 
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet001\Control\WMI\Autologger\EventLog-System\гуид в логах
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet002\Control\WMI\Autologger\EventLog-System\гуид в логах
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\WMI\Autologger\EventLog-System\гуид в логах

Открыть и там Enabled поставить в 0. Перезагрузка системы.
Также можно вырубить любую ошибку или события которые флудят но на стабильность или скорость не влияют.
